apps/src/music/utils/Loader.ts

```diff
@@ -18,69 +25,86 @@ export const LevelSources = {
 };
 
 // Loads a sound library JSON file.
-export const loadLibrary = async () => {
+export const loadLibrary = async (): Promise<MusicLibrary> => {
   if (AppConfig.getValue('local-library') === 'true') {
     const localLibraryFilename = 'music-library';
     const localLibrary = require(`@cdo/static/music/${localLibraryFilename}.json`);
-    return localLibrary;
+    return new MusicLibrary(localLibrary as LibraryJson);
   } else {
     const libraryParameter = AppConfig.getValue('library');
     const libraryFilename = libraryParameter
       ? `music-library-${libraryParameter}.json`
       : 'music-library.json';
-    const response = await fetch(baseUrl + libraryFilename);
-    const library = await response.json();
-    return library;
+
+    const libraryJsonResponse = await getJson<LibraryJson>(
+      baseUrl + libraryFilename,
+      {},
+      LibraryValidator
+    );
+    return new MusicLibrary(libraryJsonResponse.value);
```

Comment:
I opted to always handle errors in MusicView because we also need to dispatch a redux action to set the page error state. In order to avoid duplicating error handling code, I thought it would be better to just handle all errors at one level